packages/client/auth-core/src/services/CrossmintService.ts

```diff
@@ -17,36 +10,24 @@ export class CrossmintService {
         production: CROSSMINT_PROD_URL,
     };
 
-    constructor(apiKey: string, jwtToken: string | null, environment?: CrossmintEnvironment) {
+    public readonly issuer: string;
+
+    constructor(public apiKey: string, public jwtToken: string | null) {
         this.crossmintAPIHeaders = {
             accept: "application/json",
             "content-type": "application/json",
-            authorization: `Bearer ${jwtToken}`,
             "x-api-key": apiKey,
         };
-        this.crossmintBaseUrl = this.getUrlFromEnv(environment ?? "production");
-    }
 
-    async fetchCrossmintAPI({ endpoint, options = { method: "GET" }, onServerErrorMessage }: FetchCrossmintParams) {
-        const url = `${this.crossmintBaseUrl}/api/${endpoint}`;
-        const { body, method } = options;
-
-        try {
-            const response = await fetch(url, {
-                body: body ? JSON.stringify(body) : undefined,
-                method,
-                headers: this.crossmintAPIHeaders,
-            });
-            if (!response.ok) {
-                // We forward all 4XX errors. This includes rate limit errors.
-                // It also includes chain not found, as it is a bad request error.
-                throw new Error(onServerErrorMessage ?? (await response.text()));
-            }
-            return await response.json();
-        } catch (error) {
-            console.error("Error fetching Crossmint API: ", error);
-            throw new Error(`Error fetching Crossmint API: ${error}`);
-        }
+        this.crossmintBaseUrl =
+            getEnvironmentForKey(apiKey) === "production"
+                ? this.getUrlFromEnv("production")
+                : this.getUrlFromEnv("staging");
+
+        this.issuer =
```

Comment:
with my PR, this should always equal the `crossmintBaseUrl`

Reply:
hmm now that you mentioned, is that really what we want? I believe this should have two values, the crossmint or the STG one. And when developing locally or using the demo, we could use the staging one. 

If not, using the crossmintBaseUrl could lead to tokens with random issuers when developing locally, which I don't see the gain on that

Reply:
The point of the issuer is that the signer of the JWT matches the place where you use it.
The only place where this would break is if you take a local JWT and try to use it in STG / PROD or any other env.

Reply:
Leaving the comment here for visibility:
We'll have 3 issuers going forward:
- https://www.crossmint.com/ PROD
- https://staging.crossmint.com/ STG
- https://dev.crossmint.com/ LOCAL - PREVIEW

The SDK may not validate issuer anyway, as the different environments will never share keys, so public key validation should suffice.

Reply:
Perfect!